azurerm/internal/services/resource/subscription_feature_resource.go

```diff
@@ -0,0 +1,197 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+)
+
+const (
+	Pending       = "Pending"
+	Registering   = "Registering"
+	Unregistering = "Unregistering"
+	Registered    = "Registered"
+	NotRegistered = "NotRegistered"
+	Unregistered  = "Unregistered"
+)
+
+func resourceSubscriptionFeature() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceSubscriptionFeatureCreate,
+		Read:   resourceSubscriptionFeatureRead,
+		Delete: resourceSubscriptionFeatureDelete,
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.FeatureID(id)
+			return err
+		}),
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+
+			"provider_namespace": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+
+func resourceSubscriptionFeatureCreate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).Resource.FeaturesClient
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id := parse.NewFeatureID(subscriptionId, d.Get("provider_namespace").(string), d.Get("name").(string))
+
+	existing, err := client.Get(ctx, id.ProviderNamespace, id.Name)
+	if err != nil {
+		return fmt.Errorf("error checking for existing feature %q: %+v", id, err)
+	}
+
+	if existing.Properties != nil && existing.Properties.State != nil {
+		if strings.EqualFold(*existing.Properties.State, Pending) {
+			return fmt.Errorf("%s which requires manual approval should not be managed by terraform", id)
+		}
+		if !strings.EqualFold(*existing.Properties.State, NotRegistered) && !strings.EqualFold(*existing.Properties.State, Unregistered) {
+			return tf.ImportAsExistsError("azurerm_subscription_feature", id.ID())
+		}
+	}
+
+	resp, err := client.Register(ctx, id.ProviderNamespace, id.Name)
+	if err != nil {
+		return fmt.Errorf("error registering feature %q: %+v", id, err)
+	}
+
+	if resp.Properties != nil && resp.Properties.State != nil {
+		if strings.EqualFold(*resp.Properties.State, Pending) {
+			return fmt.Errorf("%s which requires manual approval can not be managed by terraform", id)
+		}
+	}
```

Comment:
since the Resource Provider has to be re-registered for this Feature to be usable - this shouldn't be a separate resource but must instead be embedded within [the `azurerm_resource_provider_registration` resource](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/resource_provider_registration)